webserver/script/serializers.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework import serializers
+from script.models.data import County
+from script.models.statistics import Energy
+
+import datetime
+
+class CountySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = County
+        fields = '__all__'
+
+    def validate_residents(self, residents):
+        if residents < 0:
+            raise serializers.ValidationError("Residents must be a positive.")
+        return residents
+
+
+class EnergySerializer(serializers.ModelSerializer):
+
+    def validate_year(self, year):
+        if year < 2000 or year > datetime.datetime.now().year:
+            raise serializers.ValidationError("Year must be 2000~now.")
+        return year
+    
+    def validate_month(self, month):
+        if month < 1 or month > 12:
+            raise serializers.ValidationError("Month must be 1~12.")
+        return month
```

Comment:
These validators are better suited to be on the actual model instead of the serializer. As the project evolves, there may be different routes a client may take to update the model. Things like public api, admin panels, etc... The validation being on the model DRYes all that up.
Here's an example: https://docs.djangoproject.com/en/2.2/ref/validators/

Reply:
Done. Thank you for your suggestions!